### PR TITLE
[GH-2404] chore(geopandas): Add empty cases to match test suite + fix edge cases

### LINKED ISSUE #2404: chore(geopandas): Add empty cases to match_geoseries test suite
parent https://github.com/apache/sedona/issues/2392

### PR DESCRIPTION
**Pull request by petern48**
## Did you read the Contributor Guide?

- Yes, I have read the [Contributor Rules](https://sedona.apache.org/latest/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest/community/develop/)

## Is this PR related to a ticket?

- Yes, and the PR name follows the format `[GH-XXX] my subject`. Closes #2404

## What changes were proposed in this PR?
Add empty cases to the `test_match_geopandas_series.py` suite. This should make it easier to catch edge cases, naturally without having to spend time giving an extra thorough PR review. I found myself forgetting about this (relatively common) case, which led to a follow-up [PR](https://github.com/apache/sedona/pull/2387) fix. It also helps a lot with finding existing differences in behavior between PostGIS and Sedona. A few of the bugs were fixed in this PR from the geopandas side, since it was not difficult to do so.

## How was this patch tested?
Added tests

## Did this PR include necessary documentation updates?

- Any behavior changes in this PR involved edge cases that already weren't documented (and are probably too edgy for it to make sense to document the behavior.

### CODE REVIEW

File: python/tests/geopandas/test_match_geopandas_series.py [hunk 1]
```diff
@@ -381,6 +393,12 @@ def test_explode(self):
 
     def test_to_crs(self):
         for geom in self.geoms:
+            if isinstance(geom[0], Polygon) and geom[0] == Polygon():
+                # SetSRID doesn't set SRID properly on empty polygon
+                # https://github.com/apache/sedona/issues/2403
+                # We replace it with a valid polygon as a workaround to pass the test
```
Comment (petern48) [inline python/tests/geopandas/test_match_geopandas_series.py:406]: Filed https://github.com/apache/sedona/issues/2403

File: python/tests/geopandas/test_match_geopandas_series.py [hunk 2]
```diff
@@ -1001,6 +1032,11 @@ def test_covered_by(self):
 
     def test_distance(self):
         for geom, geom2 in self.pairs:
+            if geom[0].is_empty or geom2[0].is_empty:
+                # Sedona's ST_Distance returns 0.0 instead of null
+                # when an empty geometry is involved
+                # https://github.com/apache/sedona/issues/2406
+                continue
```
Comment (petern48) [inline python/tests/geopandas/test_match_geopandas_series.py:1051]: Filed https://github.com/apache/sedona/issues/2406

File: python/tests/geopandas/test_match_geopandas_series.py [hunk 3]
```diff
@@ -445,6 +476,14 @@ def test_to_arrow(self):
         import pyarrow as pa
 
         for geom in self.geoms:
+            # LINEARRING EMPTY and LineString EMPTY
+            # result in 01EA03000000000000 instead of 010200000000000000.
+            # Sedona returns the right result, so this bug is likely in pyarrow or geoarrow
+            # Below we set the modify the failing case as a workaround to pass the test
+            # Occurs in python 3.9, but fixed by python 3.10.
+            if geom[0] in [LineString(), LinearRing()]:
+                geom[0] = LineString([(0, 0), (1, 1)])
```
Comment (petern48) [inline python/tests/geopandas/test_match_geopandas_series.py:485]: Couldn't find a PR fix in geoarrow for what exactly fixed this, but it's [fixed in 3.10](https://github.com/petern48/sedona/actions/runs/18619253114). I'm confident the error is not in Sedona since the following query returns the right result. Not worth investing more time in narrowing down what the issue was or where it was fixed, imo.

```sql
select ST_AsHEXEWKB(st_geomfromtext('linestring empty'));
-- 010200000000000000
```

**Review by jiayuasu — APPROVED** [review]
LGTM overall. Although I believe we should definitely fix those bugs in Sedona and remove these workarounds!
